src/main/java/org/opensearch/knn/index/KNNCircuitBreaker.java

```diff
@@ -77,16 +78,16 @@ public void initialize(ThreadPool threadPool, ClusterService clusterService, Cli
         this.clusterService = clusterService;
         this.client = client;
         Runnable runnable = () -> {
-            if (KNNWeight.knnIndexCache.isCacheCapacityReached() && clusterService.localNode().isDataNode()) {
-                long currentSizeKiloBytes =  KNNWeight.knnIndexCache.getWeightInKilobytes();
+            if (NativeMemoryCacheManager.getInstance().isCacheCapacityReached() && clusterService.localNode().isDataNode()) {
```

Comment:
nit: refactor to variable to avoid repeating getInstance?

Reply:
Sure will fix